### PR TITLE
TRI-243/Implement InMemoryQueue for handling webhook events income

### PR DESCRIPTION
**Pull request by luccar31**
# Pull Request

## Description
This PR adds the main webhook processing functionality, which includes a generic implementation for webhook adaptation to Tricker entities, logging in the database and a queue to manage inbound webhooks requests for every issue provider implemented. Also different strategy providers have been implemented to maintain SOLID principles as much as possible.
Webhooks requests that are not supported are saved and flagged for future processing.

## Ticket Link
[TRI-243](https://linear.app/tricker-sirius/issue/TRI-243/implement-inmemoryqueue-for-handling-webhook-events-income)

## Steps to Reproduce
1. Rebuild docker image and run it (`docker compose up --build`)
2. Integrate a project and go to Linear -> Workspace settings -> API and see that webhooks has been successfully registered
3. Try to make changes in Linear and watch logging of processing in stdout and in the db.

### CODE REVIEW

File: src/domains/webhook/dto/index.ts [hunk 1]
```diff
@@ -2,12 +2,10 @@
  * Interface representing the full payload of a webhook event.
  */
 export interface WebhookFullPayload {
-  /** Unique identifier of the webhook. */
-  webhookId: string;
   /** Type of event associated with the webhook payload. */
   eventType: WebhookPayloadType;
   /** Body of the webhook payload. */
-  webhookBody: JSON;
+  webhookBody: any;
```
Comment (Nachoferra1894) [inline src/domains/webhook/dto/index.ts:8]: Why is this any?

File: tests/domains/webhook/webhook.processor.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const requestMock = mock<Request>();
+
+    try {
+      await webhookProcessor.process(requestMock, 'UNSUPPORTED_PROVIDER');
+    } catch (e: any) {
+      expect(e).toBeInstanceOf(NotImplementedException);
+    }
+  });
+  //
+  // it('should throw NotImplementedException for unsupported event type in prepareWebhookPayload', async () => {
```
Comment (Nachoferra1894) [inline tests/domains/webhook/webhook.processor.test.ts:74]: This code should be uncommented or deleted

**Review by Nachoferra1894 — APPROVED** [review]

File: src/utils/errors.ts [hunk 3]
```diff
@@ -77,6 +78,25 @@ export class WebhookValidationException extends HttpException {
...
 
+export class NotImplementedException extends HttpException {
+  constructor(message?: string) {
+    super(HttpStatus.NOT_IMPLEMENTED, `Not implemented. ${message}`);
+  }
+}
+
+export class WebhooksProcessingException extends Error {
+  webhookSource: WebhookSourceDTO;
+  payload: any;
```
Comment (Nachoferra1894) [inline src/utils/errors.ts]: Why is this any? Is it possible to type the payload?

File: src/domains/webhook/dto/index.ts (hunk 1 above)
Comment (Nachoferra1894) [inline src/domains/webhook/dto/index.ts]: Same here with any

File: tests/domains/webhook/webhook.processor.test.ts (hunk 2 above)
Comment (Nachoferra1894) [inline tests/domains/webhook/webhook.processor.test.ts]: This code should be uncommented or deleted

**Review by Nachoferra1894 — CHANGES_REQUESTED** [review]

File: src/utils/errors.ts (hunk 3 above)
Comment (luccar31) [reply src/utils/errors.ts]: Fixed.

Comment (luccar31) [reply src/utils/errors.ts]: Fixed.

File: src/domains/webhook/dto/index.ts (hunk 1 above)
Comment (luccar31) [reply src/domains/webhook/dto/index.ts]: Fixed.

File: tests/domains/webhook/webhook.processor.test.ts (hunk 2 above)
Comment (luccar31) [reply tests/domains/webhook/webhook.processor.test.ts]: Fixed.

**Review by Nachoferra1894 — APPROVED** [review]

**Pull request merged**